src/main/java/org/elasticsearch/index/shard/ShadowIndexShard.java

```diff
@@ -108,7 +106,7 @@ public void updateRoutingEntry(ShardRouting newRouting, boolean persistState) {
     }
 
     @Override
-    protected Engine newEngine() {
+    protected Engine newEngine(boolean skipInitialTranslogRecovery) {
         assert this.shardRouting.primary() == false;
```

Comment:
I might be missing something (in which case, please educate me) but a shadow shard can never go through a gateway recovery, in which case we always skip the translog?